addons/point_of_sale/static/tests/unit/related_models_test.js

```diff
@@ -0,0 +1,816 @@
+            n3.update({ parent_ids: [["unlink", n2]] });
+            assert.notOk(n3.parent_ids.includes(n2), "Note should be unlinked from the old parent");
+            assert.notOk(n2.child_ids.includes(n3), "Note should be unlinked from the old child");
+        });
+
+        QUnit.test("update operation, unlink many2many", (assert) => {
+            const n1 = models["note.note"].create({});
+            const n2 = models["note.note"].create({});
+
+            n2.update({ parent_ids: [["link", n1]] });
+            assert.ok(n2.parent_ids.includes(n1), "Note should be linked to the tag");
+            assert.ok(n1.child_ids.includes(n2), "Note should be linked to the tag");
+
+            n2.update({ parent_ids: [["unlink", n1]] });
+            assert.notOk(n2.parent_ids.includes(n1), "Note should be unlinked from the tag");
+            assert.notOk(n1.child_ids.length > 0, "Note should be unlinked from the tag");
+            assert.ok(n2, "Note should be linked to the tag");
+        });
+
+        QUnit.test("update operation, Clear many2many", (assert) => {
+            const note = models["note.note"].create({});
+            const note2 = models["note.note"].create({});
+            const note3 = models["note.note"].create({
+                name: "note 1",
+                parent_ids: [["link", note, note2]],
+            });
+            assert.equal(note3.parent_ids.length, 2, "note 3 should have parent notes");
+            models["note.note"].update(note3, { parent_ids: [["clear"]] });
+            assert.equal(
+                note3.parent_ids.length,
+                0,
+                "All parent notes should be unlinked from the note"
+            );
+            models["note.note"].update(note, { child_ids: [["clear"]] });
+            assert.equal(
+                note.child_ids.length,
+                0,
+                "All child notes should be unlinked from the note"
+            );
+        });
+
+        QUnit.test("delete operation, many2many item", (assert) => {
+            const n1 = models["note.note"].create({});
+            const n2 = models["note.note"].create({});
+            const n3 = models["note.note"].create({});
+            n3.update({ parent_ids: [["link", n1, n2]] });
+
+            assert.ok(
+                n3.parent_ids.includes(n1, n2),
+                "Note should be linked to the parent nodes 1 and 2"
+            );
+
+            n1.delete();
+            assert.notOk(models["note.note"].read(n1.id), "Note should be deleted");
+            assert.notOk(n3.parent_ids.includes(n1), "Note 1 should be unlinked from note 3");
+            n1.update({ child_ids: [["link", n3]] });
+
+            assert.ok(n3.parent_ids.includes(n1), "Note 3 should be linked to note 1");
+
+            n3.delete();
+            assert.notOk(models["note.note"].read(n3.id), "Tag should be deleted");
+            assert.notOk(n1.child_ids.length > 0, "Note should be unlinked from the tag");
```

Comment:
```suggestion
```
relinking deleted records shouldn't be allowed. We'll do this check on a different task.